src/scriptProfiler.cpp

```diff
@@ -1215,9 +1240,72 @@ class GameInstructionNewExpression : public game_instruction {
 
 #pragma endregion Instructions
 #endif
+
+
+#if __linux__
+static std::filesystem::path sharedObjectPath;
+extern "C" int iterateCallback(struct dl_phdr_info* info, size_t size, void* data) {
+    std::filesystem::path sharedPath = info->dlpi_name;
+    if (sharedPath.filename() == "ArmaScriptProfiler_x64.so"sv) {
+        sharedObjectPath = sharedPath;
+        return 0;
+    }
+    return 0;
+}
+#endif
+
+std::filesystem::path getSharedObjectPath() {
+#if __linux__
+    dl_iterate_phdr(iterateCallback, nullptr);
+    return sharedObjectPath;
+
+#else
+    wchar_t buffer[MAX_PATH] = { 0 };
+    HMODULE handle = nullptr;
+    if(GetModuleHandleExW(GET_MODULE_HANDLE_EX_FLAG_FROM_ADDRESS | GET_MODULE_HANDLE_EX_FLAG_UNCHANGED_REFCOUNT,
+            reinterpret_cast<LPCWSTR>(&getSharedObjectPath), &handle) == 0) {
+        sqf::diag_log("getSharedObjectPath: GetModuleHandle failed");
+        return std::filesystem::path{};
+    }
+
+    DWORD len = GetModuleFileNameW(handle, buffer, MAX_PATH);
+    if (len == 0) {
+        return std::filesystem::path{};
+    }
+    return std::filesystem::path{buffer};
+#endif
+}
+
+std::filesystem::path findConfigFilePath() {
+
+    std::filesystem::path path = getSharedObjectPath();
+    path = path.parent_path().parent_path();

```

Comment:
This will be weird with nested mod folders like `arma3server/server_mods/my_favourite_ones/@asp`. Better to look inside `@asp` folder i.e. just one parent_path to also avoid bloating the arma server folder. Will also work better with server providers that symlinks the mod folders together into a temp folder.

Reply:
Yeah, it returns the full path to the plugin. To get the ASP folder we need to back out of file and the intercept directory. Yeah, it is a little cleaner to use the parent_path function rather than adding the "go up path" string and having to make the path normalized.